src/machine/asm/mod.rs

```diff
@@ -54,6 +54,27 @@ impl CoreMachine for Box<AsmCoreMachine> {
     }
 }
 
+fn inited_memory(machine: &mut AsmCoreMachine, addr: u64, size: u64) -> Result<(), Error> {
+    let frame_from = addr / MEMORY_FRAMESIZE as u64;
+    let (addr_to, overflowed) = addr.overflowing_add(size);
+    if overflowed {
+        return Err(Error::OutOfBound);
+    }
+    let frame_to = addr_to / MEMORY_FRAMESIZE as u64;
+    for i in frame_from..=std::cmp::min(MEMORY_FRAMES as u64 - 1, frame_to) {
+        if machine.frames[i as usize] == 0 {
+            let base_addr = i * MEMORY_FRAMESIZE as u64;
+            memset(
+                &mut machine.memory
+                    [base_addr as usize..(base_addr + MEMORY_FRAMESIZE as u64) as usize],
+                0,
+            );
+            machine.frames[i as usize] = 1;
+        }
+    }
+    Ok(())
+}
+
```

Comment:
Since we check for out-of-bounds here, can we remove these code blocks below?
```
        if addr + 2 > self.memory.len() as u64 {
            return Err(Error::OutOfBound);
        }

        if addr + 4 > self.memory.len() as u64 {
            return Err(Error::OutOfBound);
        }
```

suggest to cast to usize instead of u64

```suggestion
    let (addr_to, overflowed) = addr.overflowing_add(size);
    if overflowed {
        return Err(Error::OutOfBound);
    }
    let frame_from = addr as usize / MEMORY_FRAMESIZE;
    let frame_to = addr_to as usize / MEMORY_FRAMESIZE;
    for i in frame_from..=std::cmp::min(MEMORY_FRAMES - 1, frame_to) {
        if machine.frames[i] == 0 {
            let base_addr = i * MEMORY_FRAMESIZE;
            memset(
                &mut machine.memory[base_addr..(base_addr + MEMORY_FRAMESIZE)],
                0,
            );
            machine.frames[i] = 1;
        }
    }
```

Reply:
> Since we check for out-of-bounds here, can we remove these code blocks below ?

inited_memory() does not check out-of-bounds.

As https://github.com/nervosnetwork/ckb-vm/pull/107/files#r469094023 says, I will combine these logics: `inited_memory`, `out-of-bounds-check` and `check_write_permission`

Reply:
I do suggest we keep u64 here to be explicit with the types, and only cast it to usize when we do need indexing. ckb-vm has employed this way throughout the codebase to be more precise with types.